doc/priorityClass.md

```diff
@@ -0,0 +1,46 @@
+# PriorityClass proposed flow design
+
+This page describes the design and implementation of the PriorityClass
+support in the compliance operator.
+
+## Goals
+We want to accomplish:
+   * User can set PriorityClass in ScanSetting
+   * Pods will be launched with the PriorityClass set in ScanSetting
+
+## Add PriorityClass option to ScanSetting CRD
+
+Administrators can choose which PriorityClass in the ScanSetting.
+A sample looks as follows:
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: my-companys-constraints
+autoApplyRemediations: false
+autoUpdateRemediations: false
+schedule: "0 1 * * *"
+priorityClass: my-priority-class
+rawResultStorage:
+  size: "2Gi"
+  rotation: 10
+# For each role, a separate scan will be created pointing
+# to a node-role specified in roles
+roles:
+  - worker
+  - master
+```
+
+We will be adding priorityClass to ScanSetting CRD as a optional field,
+and if none is specified, we will treat as it didn't have a PriorityClass.
+Otherwise, we will use the PriorityClass specified in the ScanSetting.
+
+## Launch related scanning pod with set PriorityClass in ScanSetting
+
+Once an admin set a PriorityClass in ScanSetting, when this ScanSetting
+is used, we will try to launch all the pod with that PriorityClass.
+
+## Questions
+
+Will we need to have a default priorityClass for compliance operator pod, and such?
```

Comment:
I don't think we need a default priority class for the Compliance Operator manager pod itself. It's not critical to have it running all the time and should be tolerant of being pre-empted. All that would happen is that scans would take longer to have an up-to-date status. What's critical is to have scans, results, and aggregations happen. Those would need an appropriate priorityclass set.

Reply:
that's a good point! thanks for taking a look. I will just use PriorityClass on all scanning pods.

Reply:
When you say scanning pods, do you really mean just scanners or all the "support" pods like resultServer, aggregator or such?

Reply:
> When you say scanning pods, do you really mean just scanners or all the "support" pods like resultServer, aggregator or such?

Yes, all the related pods

Reply:
We won't offer the ability to define PriorityClasses for different pods per scan, right (e.g., always make sure the resultsServer has a high PriorityClass)?

Reply:
that's right, all pods should use the same PriorityClass per scan.